object-search/app.py

```diff
@@ -4,44 +4,42 @@
 import os
 import shutil
 import sys
+from glob import glob
 import requests
 
 import click
 from jina.flow import Flow
 from jina import Document
 from jina.clients.sugary_io import _input_lines
 from jina.logging import default_logger as logger
+from jina.logging.profile import TimeContext
 
-
-num_docs = os.environ.get('MAX_DOCS', 16)
-data_path = 'data/**/*.jpg'
-batch_size = 16
-
-
-def clean_workdir():
-    if os.path.exists(os.environ['WORKDIR']):
-        shutil.rmtree(os.environ['WORKDIR'])
-        logger.warning('Workspace deleted')
+MAX_DOCS = int(os.environ.get("JINA_MAX_DOCS", 50))
+IMAGE_SRC = 'data/f8k/images/*.jpg'
 
 
 def config():
-    parallel = 1 if sys.argv[1] == 'index' else 1
-    shards = 1
-
-    os.environ['PARALLEL'] = str(parallel)
-    os.environ['SHARDS'] = str(shards)
+    os.environ["JINA_DATA_FILE"] = os.environ.get("JINA_DATA_FILE", IMAGE_SRC)
+    os.environ['PARALLEL'] = str(1)
+    os.environ['SHARDS'] = str(1)
+    os.environ["JINA_PORT"] = os.environ.get("JINA_PORT", str(45678))
     os.environ['WORKDIR'] = './workspace'
-    os.makedirs(os.environ['WORKDIR'], exist_ok=True)
-    os.environ['JINA_DATA_FILE'] = os.environ.get('JINA_DATA_FILE', 'data/**/*.jpg')
-    os.environ['JINA_PORT'] = os.environ.get('JINA_PORT', str(45678))
+
+
+def index(num_docs: int):
+    f = Flow.load_config('flow-index.yml')
+    num_docs = min(num_docs, len(glob(IMAGE_SRC)))
+    with f:
+        with TimeContext(f'QPS: indexing {num_docs}', logger=f.logger):
+            f.index_files(IMAGE_SRC, request_size=10, read_mode='rb', size=num_docs)
 
 
 def index_restful(num_docs):
     f = Flow().load_config('flow-index.yml')
 
     with f:
         data_path = os.path.join(os.path.dirname(__file__), os.environ.get('JINA_DATA_FILE', None))
-        print(f'Indexing {data_path}')
+        logger.info(f'Indexing {data_path}')
```

Comment:
I am actually not sure here, but maybe its better to use `f.logger` here instead of the default logger? Same for the other logger usages where we have the flow object already. On the other hand the flow is not available yet in main and you need to use some other logger there.

Reply:
https://github.com/jina-ai/examples/pull/477/commits/d83338770de9b7fd5282cc2f12fda47cd0381be9